src/alchemlyb/parsing/amber.py

```diff
@@ -288,7 +295,10 @@ def extract(outfile, T):
                 mbar = secp.extract_section('^MBAR', '^ ---', file_datum.mbar_lambdas,
                                             extra=line)
                 
-                if None in mbar: # pragma: no cover
+                if None in mbar:
+                    msg = "WARNING, something strange parsing the following MBAR section."
+                    msg += "\nMaybe the mbar_lambda values are incorrect?"
+                    logger.warning(f"{msg}\n{mbar}")
```

Comment:
Makes me shudder that we just continue here instead of failing. We obviously had this code here before but it makes me uncomfortable that we allow continuing and potentially creating wrong results. Or would this be considered a normal situation when working with AMBER files??

Reply:
As far as I know, this is not considered a normal situation with AMBER files, I kept things as they were, but I agree a proper exception is better suited here.

Reply:
Let's raise an exception here. Use `ValueError` whenever something is wrong in the input file because we use `ValueError` in the NAMD parser in the same context. Please add a test.

Reply:
Done and test added